sortinghat/core/api.py

```diff
@@ -503,3 +504,107 @@ def withdraw(uuid, organization, from_date=None, to_date=None):
     uidentity.refresh_from_db()
 
     return uidentity
+
+
+@django.db.transaction.atomic
+def merge_identities(from_uuid, to_uuid):
+    """
+    Merge one unique identity into another.
+
+    Use this function to join 'from_uuid' unique identity into
+    'to_uuid'. Identities and enrollments related to 'from_uuid' will be
+    assigned to 'to_uuid'. In addition, 'from_uuid' will be removed
+    from the registry. Duplicated enrollments will be also removed from
+    the registry while overlapped enrollments will be merged.
+
+    This function also merges two profiles. When a field on 'to_uuid'
+    profile is None or empty, it will be updated with the value on the
+    profile of 'from_uuid'. If any of the two unique identities was set
+    as a bot, the new profile will also be set as a bot.
+
+    When 'from_uuid' and 'to_uuid' are equal, or any of this two fields
+    are 'None' or empty, it raises an 'InvalidValueError'.
+
+    The function raises a 'NotFoundError exception when either 'from_uuid'
+    or 'to_uuid' do not exist in the registry.
+
+    :param from_uuid: identifier of the unique identity set to merge
+    :param to_uuid: identifier of the unique identity where 'from_uuid'
+        will be merged
+
+    :raises NotFoundError: raised when either 'from_uuid' or 'to_uuid'
+        do not exist in the registry
+    """
+    if from_uuid is None:
+        raise InvalidValueError(msg="'from_uuid' cannot be None")
+    if from_uuid == '':
+        raise InvalidValueError(msg="'from_uuid' cannot be an empty string")
+    if to_uuid is None:
+        raise InvalidValueError(msg="'to_uuid' cannot be None")
+    if to_uuid == '':
+        raise InvalidValueError(msg="'to_uuid' cannot be an empty string")
+    if from_uuid == to_uuid:
+        raise InvalidValueError(msg="'from_uuid' and 'to_uuid' cannot be equal")
+
+    try:
+        from_uid = find_unique_identity(from_uuid)
+        to_uid = find_unique_identity(to_uuid)
+    except NotFoundError as exc:
+        # Identities were not found, so they cannot be merged
+        raise exc
+
+    try:
+        identities = from_uid.identities.all()
+    except ValueError as e:
+        raise InvalidValueError(msg=str(e))
+
+    uidentity = to_uid
+    for identity in identities:
+        uidentity = move_identity_db(identity, to_uid)
```

Comment:
Yes. This was because first I didn't want to modify the value of `to_uid` variable, but after your comment I think it does not make sense to have `uidentity` as a variable. Now, only `to_uid` is used.